Add io.github.itprojects.MasVisGtk

itprojects opened the pull request:
<!-- ⚠️  The submission PR must be against the `new-pr` branch ⚠️  -->

### Please confirm your submission meets all the criteria

<!-- Please replace each `[ ]` with `[X]` when the step is complete -->

- [X] Please describe the application briefly.

MasVisGtk is an audio file (loudness) analysis UI application, using numpy, scipy, pyplot (matplotlib).

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

**NOTE**: the application started as a fork of a GPL 2 licensed python app/module, but grew into a full, and distinct application. The repository maintainer hasn't had any activity on their page for **2 years**.


<!-- ⚠️  Don't modify anything below this line ⚠️  -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

hfiguiere commented on io.audio.MasVisGtk.json:
```diff
@@ -0,0 +1,389 @@
+{
+    "id" : "io.audio.MasVisGtk",
```
`audio.io` is a parked domain.

Yet you acknowledged:

> The domain used for the application ID is [controlled by the application developers](https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository) and the [application id guidelines](https://docs.flathub.org/docs/for-app-authors/requirements#application-id) are followed.

itprojects replied:
Is it?

itprojects replied:
What about io.music.MasVisGtk ?

hfiguiere replied:
![image](https://github.com/user-attachments/assets/b074946b-8f56-4d85-9560-1f1ebea710a8)

hfiguiere replied:
> What about io.music.MasVisGtk ?

One doesn't randomly pick a domain. You can use the one based on the github project as per the documentation.

hfiguiere replied:
![image](https://github.com/user-attachments/assets/057c183f-056f-43db-9538-5da104ff980a)

hfiguiere replied:
And just to clarify. Upstream choice is wrong.

itprojects replied:
How does one choose a namespace that doesn't have github or a website in it?

itprojects replied:
For example audioanalysis.ui.MasVisGtk

bbhtt replied:
> How does one choose a namespace that doesn't have github or a website in it?

You or the project own or control the domain

hfiguiere replied:
`aud` is a valid top-level-domain.

I am not sure what you don't understand with :

> The domain used for the application ID is [controlled by the application developers](https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository) and the [application id guidelines](https://docs.flathub.org/docs/for-app-authors/requirements#application-id) are followed.

But just to debunk any fiction: a VALID domain, CONTROLLED by the UPSTREAM developers. If there is no such thing, then something based on the github project URL as per the documentation.

There is no alternative.

itprojects replied:
If the app is altered to be in namespace `io.github.itprojects.MasVisGtk` will this PR have to be resubmitted, or could a new branch be used?

hfiguiere replied:
the branch name is irrelevant.

itprojects replied:
The domain is changed to `io.github.itprojects.MasVisGtk`.

hfiguiere commented on io.audio.MasVisGtk.json:
```diff
@@ -0,0 +1,389 @@
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--filesystem=home:create"
```
```suggestion
        "--filesystem=home"
```

itprojects replied:
Would that not stop saving of analysis files?

hfiguiere replied:
what does "create home directory" means to you?

itprojects replied:
Exactly. It will be fixed.

bbhtt commented on io.audio.MasVisGtk.json:
```diff
@@ -0,0 +1,389 @@
+        "*.a"
+    ],
+    "modules" : [
+		{
```
These python sources should be in a separate file since you auto-generate them

bbhtt commented on io.audio.MasVisGtk.json:
```diff
@@ -0,0 +1,389 @@
+				    "url": "https://files.pythonhosted.org/packages/38/4f/8487737a74d8be4ab5fbe6019b0fae305c1604cf7209500969b879b5f462/matplotlib-3.8.4.tar.gz",
+				    "sha256": "8aac397d5e9ec158960e31c381c5ffc52ddd52bd9a47717e2a694038167dffea"
+				},
+				{
+					"type": "archive",
+					"url": "https://github.com/qhull/qhull/archive/refs/tags/2020.2.tar.gz",
+					"dest": "build/qhull-2020.2",
+					"sha256": "59356b229b768e6e2b09a701448bfa222c37b797a84f87f864f97462d8dbc7c5"
+				}
```
This should be in the main manifest.

bbhtt commented on io.audio.MasVisGtk.json:
```diff
@@ -0,0 +1,389 @@
+			"build-commands": [
+				"python3 setup.py install --prefix=/app --root=/"
+			],
+			"sources": [
```
qhull and freetype should be in the main manifest

bbhtt commented on io.audio.MasVisGtk.json:
```diff
@@ -0,0 +1,389 @@
+			]
+		},
+		{
+			"name": "python3-openblas-lapack",
```
This should be in the main manifest

itprojects replied:
This may be a problem. `numpy`, (and maybe) `scipy` and `matplotlib` need to have this with them, for optimal computation.

bbhtt replied:
I didn't tell you remove it completely, install these in the main manifest, separate all the python parts and place it _after_ these dependencies.

itprojects replied:
`python3-openblas-lapack` is a computational package, why would you put that in the main manifest, but `numpy` and the others in `python3-deps.json`?

bbhtt replied:
Because neither of blas or lapack are being generated by flatpak-pip-generator. These are system dependencies.

itprojects replied:
As you know, if the order is wrong, the package may not compile at all. So doees that mean 16 json files?
Then to make it simple, which of the following packages goes to a separate file `python3-deps` ?

```
python3-meson-python
python3-typing-extensions
python3-setuptools-scm
python3-pybind11
python3-openblas-lapack
python3-numpy
python3-matplotlib-deps
    python3-contourpy
    python3-flit-core
    python3-tomli
    python3-certifi
    python3-cppy
    python3-Pillow
    python3-cycler
    python3-kiwisolver
    python3-pyparsing
    python3-python_dateutil
python3-pythran
python3-scipy
python3-matplotlib
```

bbhtt replied:
No in your main manifest:

```json
				{
				    "name": "openblas",
				    "no-autogen": true,
				    "make-args": [
				        "DYNAMIC_ARCH=1",
				        "FC=gfortran",
				        "NO_CBLAS=1",
				        "NO_LAPACKE=1",
				        "USE_OPENMP=0"
				    ],
				    "make-install-args": [
				        "PREFIX=/app"
				    ],
				    "sources": [
				        {
				            "type": "archive",
				            "url": "https://github.com/OpenMathLib/OpenBLAS/archive/refs/tags/v0.3.28.tar.gz",
				            "sha256": "f1003466ad074e9b0c8d421a204121100b0751c96fc6fcf3d1456bd12f8a00a1"
				        }
				    ]
				},
				{
				    "name": "lapack",
				    "buildsystem": "cmake-ninja",
				    "builddir": true,
				    "cleanup": [
				        "/lib/cmake"
				    ],
				    "config-opts": [
				        "-DCMAKE_BUILD_TYPE=RelWithDebInfo",
				        "-DBUILD_SHARED_LIBS=ON",
				        "-DBUILD_TESTING=OFF",
				        "-DLAPACKE=ON",
				        "-DCBLAS=ON"
				    ],
				    "sources": [
				        {
				            "type": "archive",
				            "url": "https://github.com/Reference-LAPACK/lapack/archive/refs/tags/v3.12.0.tar.gz",
				            "sha256": "eac9570f8e0ad6f30ce4b963f4f033f0f643e7c3912fc9ee6cd99120675ad48b"
				        }
				    ]
				}
			]
		},
    {
			"type": "archive",
			"url": "https://gitlab.freedesktop.org/freetype/freetype/-/archive/VER-2-6-1/freetype-VER-2-6-1.tar.gz",
			"sha256": "85764250282ec0b8ff2918f48a5c462be4d2089456e3e1d65f1bd6648cb5000c"
     },
		{
			"type": "archive",
			"url": "https://github.com/qhull/qhull/archive/refs/tags/2020.2.tar.gz",
				"dest": "build/qhull-2020.2",
				"sha256": "59356b229b768e6e2b09a701448bfa222c37b797a84f87f864f97462d8dbc7c5"
		},
    "python3-deps.json"
```

where python3-deps contains only the python packages you generate with flatpak-pip-generator.

itprojects replied:
What about `qhull` and `freetype`?

bbhtt replied:
Add them after lapack

itprojects replied:
How do you build them. There seem to be errors, when moving them as their own module {}. Doesn't `matplotlib` want them inside so that they are "baked into the code"?

```
========================================================================
Building module freetype in /home/<USERNAME>/.var/app/org.gnome.Builder/cache/gnome-builder/flatpak-builder/build/freetype-1
========================================================================
Running: python3 setup.py install --prefix=/app --root=/
python3: can't open file '/run/build/freetype/setup.py': [Errno 2] No such file or directory
Error: module freetype: Child process exited with code 2
```

Do you know how to build the latest `matplotlib=3.9.2`, `numpy`, `scipy` ?

bbhtt replied:
Freetype should be in the runtime already, you don't have to build it here. Just pass
`-Dsystem-freetype=true` to matplotlib

See https://github.com/flathub/org.artisan_scope.artisan/blob/9d5f356d7833549d99de9a60446c437587c1bb9e/dep-python3-source.json#L10

itprojects replied:
`freetype` and `qhull` seem to compile with your good advice.

`openblas` is in the main json.

`python3-scipy` from `python3-deps.json` fails 

```
		{
		    "name": "python3-scipy",
		    "buildsystem": "simple",
		    "build-commands": [
		        "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"scipy\" --no-build-isolation"
		    ],
		    "sources": [
		        {
		            "type": "file",
		            "url": "https://files.pythonhosted.org/packages/30/85/cdbf2c3c460fe5aae812917866392068a88d02f07de0fe31ce738734c477/scipy-1.12.0.tar.gz",
		            "sha256": "4bf5abab8a36d20193c698b0f1fc282c1d083c94723902c447e5d2f1780936a3"
		        }
		    ]
		},
```

Error

```
========================================================================
Building module python3-scipy in /home/<USERNAME>/.var/app/org.gnome.Builder/cache/gnome-builder/flatpak-builder/build/python3-scipy-1
========================================================================
Running: pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} "scipy" --no-build-isolation
Using pip 24.1.2 from /usr/lib/python3.11/site-packages/pip (python 3.11)
Looking in links: file:///run/build/python3-scipy
Processing ./scipy-1.12.0.tar.gz
  Running command Preparing metadata (pyproject.toml)
  + meson setup /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7 /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7/.mesonpy-zjne_yxy -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=/tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7/.mesonpy-zjne_yxy/meson-python-native-file.ini
  The Meson build system
  Version: 1.3.2
  Source dir: /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7
  Build dir: /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7/.mesonpy-zjne_yxy
  Build type: native build
  Project name: scipy
  Project version: 1.12.0
  C compiler for the host machine: ccache cc (gcc 13.2.0 "cc (GCC) 13.2.0")
  C linker for the host machine: cc ld.bfd 2.42
  C++ compiler for the host machine: ccache c++ (gcc 13.2.0 "c++ (GCC) 13.2.0")
  C++ linker for the host machine: c++ ld.bfd 2.42
  Cython compiler for the host machine: cython (cython 0.29.37)
  Host machine cpu family: x86_64
  Host machine cpu: x86_64
  Program python found: YES (/usr/bin/python)
  Found pkg-config: YES (/usr/bin/pkg-config) 2.2.0
  Run-time dependency python found: YES 3.11
  Program cython found: YES (/usr/bin/cython)
  Compiler for C supports arguments -Wno-unused-but-set-variable: YES
  Compiler for C supports arguments -Wno-unused-function: YES
  Compiler for C supports arguments -Wno-conversion: YES
  Compiler for C supports arguments -Wno-misleading-indentation: YES
  Library m found: YES
  Fortran compiler for the host machine: gfortran (gcc 13.2.0 "GNU Fortran (GCC) 13.2.0")
  Fortran linker for the host machine: gfortran ld.bfd 2.42
  Compiler for Fortran supports arguments -Wno-conversion: YES
  Checking if "-Wl,--version-script" : links: YES
  Program pythran found: YES (/app/bin/pythran)
  Found CMake: /usr/bin/cmake (3.29.6)
  WARNING: CMake Toolchain: Failed to determine CMake compilers state
  Run-time dependency xsimd found: NO (tried pkgconfig and cmake)
  Run-time dependency threads found: YES
  Library npymath found: YES
  Library npyrandom found: YES
  pybind11-config found: YES (/app/bin/pybind11-config) 2.11.1
  Run-time dependency pybind11 found: YES 2.11.1
  Run-time dependency scipy-openblas found: NO (tried pkgconfig)
  Run-time dependency openblas found: NO (tried pkgconfig and cmake)
  Run-time dependency openblas found: NO (tried pkgconfig and cmake)

  ../scipy/meson.build:169:9: ERROR: Dependency "OpenBLAS" not found, tried pkgconfig and cmake

  A full log can be found at /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7/.mesonpy-zjne_yxy/meson-logs/meson-log.txt
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /usr/bin/python /usr/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp2ivqrx0r
  cwd: /tmp/pip-install-ytc4zjg6/scipy_1ce7c2a53d454c308c1382a557048dc7
  Preparing metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Error: module python3-scipy: Child process exited with code 1
```

Do you know how to compile [scipy 1.14.1 from source](https://files.pythonhosted.org/packages/62/11/4d44a1f274e002784e4dbdb81e0ea96d2de2d1045b2132d5af62cc31fd28/scipy-1.14.1.tar.gz) and [numpy 2.1.0](https://files.pythonhosted.org/packages/54/a4/f8188c4f3e07f7737683588210c073478abcb542048cf4ab6fedad0b458a/numpy-2.1.0.tar.gz)?

bbhtt replied:
Try setting this for scipy after `buildsystem`:

```json
            "build-options": {
                "env": {
                    "ATLAS": "None",
                    "BLAS": "/app/lib",
                    "LAPACK": "/app/lib"
                }
            },
```

bbhtt replied:
what's the issue with numpy ?

itprojects replied:
The numpy version is old and slow 1.26.4.

itprojects replied:
```
		{
		    "name": "python3-scipy",
		    "buildsystem": "simple",
		    "build-options": {
                "env": {
                    "ATLAS": "None",
                    "BLAS": "/app/lib",
                    "LAPACK": "/app/lib"
                }
            },
		    "build-commands": [
		        "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"scipy\" --no-build-isolation"
		    ],
		    "sources": [
		        {
		            "type": "file",
		            "url": "https://files.pythonhosted.org/packages/30/85/cdbf2c3c460fe5aae812917866392068a88d02f07de0fe31ce738734c477/scipy-1.12.0.tar.gz",
		            "sha256": "4bf5abab8a36d20193c698b0f1fc282c1d083c94723902c447e5d2f1780936a3"
		        }
		    ]
		},
```

```
========================================================================
Building module python3-scipy in /home/<USERNAME>/.var/app/org.gnome.Builder/cache/gnome-builder/flatpak-builder/build/python3-scipy-2
========================================================================
Running: pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} "scipy" --no-build-isolation
Using pip 24.1.2 from /usr/lib/python3.11/site-packages/pip (python 3.11)
Looking in links: file:///run/build/python3-scipy
Processing ./scipy-1.12.0.tar.gz
  Running command Preparing metadata (pyproject.toml)
  + meson setup /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587 /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587/.mesonpy-4h_b9mmo -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=/tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587/.mesonpy-4h_b9mmo/meson-python-native-file.ini
  The Meson build system
  Version: 1.3.2
  Source dir: /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587
  Build dir: /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587/.mesonpy-4h_b9mmo
  Build type: native build
  Project name: scipy
  Project version: 1.12.0
  C compiler for the host machine: ccache cc (gcc 13.2.0 "cc (GCC) 13.2.0")
  C linker for the host machine: cc ld.bfd 2.42
  C++ compiler for the host machine: ccache c++ (gcc 13.2.0 "c++ (GCC) 13.2.0")
  C++ linker for the host machine: c++ ld.bfd 2.42
  Cython compiler for the host machine: cython (cython 0.29.37)
  Host machine cpu family: x86_64
  Host machine cpu: x86_64
  Program python found: YES (/usr/bin/python)
  Found pkg-config: YES (/usr/bin/pkg-config) 2.2.0
  Run-time dependency python found: YES 3.11
  Program cython found: YES (/usr/bin/cython)
  Compiler for C supports arguments -Wno-unused-but-set-variable: YES
  Compiler for C supports arguments -Wno-unused-function: YES
  Compiler for C supports arguments -Wno-conversion: YES
  Compiler for C supports arguments -Wno-misleading-indentation: YES
  Library m found: YES
  Fortran compiler for the host machine: gfortran (gcc 13.2.0 "GNU Fortran (GCC) 13.2.0")
  Fortran linker for the host machine: gfortran ld.bfd 2.42
  Compiler for Fortran supports arguments -Wno-conversion: YES
  Checking if "-Wl,--version-script" : links: YES
  Program pythran found: YES (/app/bin/pythran)
  Found CMake: /usr/bin/cmake (3.29.6)
  WARNING: CMake Toolchain: Failed to determine CMake compilers state
  Run-time dependency xsimd found: NO (tried pkgconfig and cmake)
  Run-time dependency threads found: YES
  Library npymath found: YES
  Library npyrandom found: YES
  pybind11-config found: YES (/app/bin/pybind11-config) 2.11.1
  Run-time dependency pybind11 found: YES 2.11.1
  Run-time dependency scipy-openblas found: NO (tried pkgconfig)
  Run-time dependency openblas found: NO (tried pkgconfig and cmake)
  Run-time dependency openblas found: NO (tried pkgconfig and cmake)

  ../scipy/meson.build:169:9: ERROR: Dependency "OpenBLAS" not found, tried pkgconfig and cmake

  A full log can be found at /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587/.mesonpy-4h_b9mmo/meson-logs/meson-log.txt
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /usr/bin/python /usr/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpez6wilgr
  cwd: /tmp/pip-install-mdr55mbs/scipy_9a83efb258414406a16a36dbf966e587
  Preparing metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Error: module python3-scipy: Child process exited with code 1
```

bbhtt replied:
can you push the rest of the changes until now? We can figure it out in the PR.

itprojects replied:
Pushing soon.

itprojects replied:
Pushed to one of the other branches. Have a look.

bbhtt replied:
Please push it here. I don't know where this other branch is.

itprojects replied:
Added updates to reflect changes.

bbhtt commented on io.audio.MasVisGtk.json:
```diff
@@ -0,0 +1,389 @@
+    "modules" : [
+		{
+			"name": "python3-meson-python",
+			"buildsystem": "simple",
```
meson-python should not be required at runtime, you should add a cleanup to remove it.

`"cleanup": ["*"]`

itprojects replied:
Would the follow work?

```
		{
			"name": "python3-meson-python",
			"buildsystem": "simple",
			"build-commands": [
				"pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"meson-python\" --no-build-isolation"
			],
			"sources": [
				{
				    "type": "file",
				    "url": "https://files.pythonhosted.org/packages/91/c0/104cb6244c83fe6bc3886f144cc433db0c0c78efac5dc00e409a5a08c87d/meson_python-0.16.0-py3-none-any.whl",
				    "sha256": "842dc9f5dc29e55fc769ff1b6fe328412fe6c870220fc321060a1d2d395e69e8"
				},
				{
				    "type": "file",
				    "url": "https://files.pythonhosted.org/packages/08/aa/cc0199a5f0ad350994d660967a8efb233fe0416e4639146c089643407ce6/packaging-24.1-py3-none-any.whl",
				    "sha256": "5b8f2217dbdbd2f7f384c41c628544e6d52f2d0f53c6d0c3ea61aa5d1d7ff124"
				},
				{
				    "type": "file",
				    "url": "https://files.pythonhosted.org/packages/aa/5f/bb5970d3d04173b46c9037109f7f05fc8904ff5be073ee49bb6ff00301bc/pyproject_metadata-0.8.0-py3-none-any.whl",
				    "sha256": "ad858d448e1d3a1fb408ac5bac9ea7743e7a8bbb472f2693aaa334d2db42f526"
				}
			],
			"cleanup": ["*"]
		},
```

bbhtt replied:
Yes

hfiguiere commented on com.github.itprojects.MasVisGtk.json:
```diff
@@ -0,0 +1,99 @@
+{
+    "id" : "com.github.itprojects.MasVisGtk",
```
the document clearly say that `io.github` should be used in that case.

bbhtt replied:
Please follow the docs, not what other apps are doing. Those apps were submitted before the policy.

itprojects replied:
So policies _CAN_ change.

itprojects replied:
A new code version exists (compiles and runs latest deps), but it seems it will have to be refactored.

bbhtt replied:
> So policies _CAN_ change.

Yes

bbhtt commented on com.github.itprojects.MasVisGtk.json:
```diff
@@ -0,0 +1,99 @@
+            ]
+		},
+		"python3-deps.json",
+        {
```
indentation is broken here

bbhtt commented on com.github.itprojects.MasVisGtk.json:
```diff
@@ -0,0 +1,99 @@
+		        "url": "https://github.com/qhull/qhull/archive/refs/tags/2020.2.tar.gz",
+		        "sha256": "59356b229b768e6e2b09a701448bfa222c37b797a84f87f864f97462d8dbc7c5"
+		        }
+		    ],
```
indentation is broken here

bbhtt commented on com.github.itprojects.MasVisGtk.json:
```diff
@@ -0,0 +1,99 @@
+                    "url":"https://github.com/Reference-LAPACK/lapack/archive/refs/tags/v3.12.0.tar.gz",
+                    "sha256":"eac9570f8e0ad6f30ce4b963f4f033f0f643e7c3912fc9ee6cd99120675ad48b"
+                }
+            ]
+        },
+		{
```
indentation is broken here

bbhtt commented on com.github.itprojects.MasVisGtk.json:
```diff
@@ -0,0 +1,99 @@
+            	"-DBUILD_STATIC_LIBS=OFF"
+            ],
+		    "sources": [
+		        {
+		        "type": "archive",
```
indentation is broken here

bbhtt commented on com.github.itprojects.MasVisGtk.json:
```diff
@@ -0,0 +1,99 @@
+            "name":"lapack",
+            "buildsystem":"cmake-ninja",
+            "builddir":true,
+            "cleanup":[
+            "/lib/cmake"
+            ],
```
indentation is broken here

bbhtt commented on python3-deps.json:
```diff
@@ -0,0 +1,305 @@
+		    "name": "python3-scipy",
+		    "buildsystem": "simple",
+		    "build-options": {
+                "env": {
+                    "ATLAS": "None",
```
indentation is broken here

bbhtt commented on python3-deps.json:
```diff
@@ -0,0 +1,305 @@
+		{
+		    "name": "python3-matplotlib",
+		    "buildsystem": "simple",
+		    "build-commands": [
+		                        "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"matplotlib==3.9.2\" --no-build-isolation --config-settings=setup-args=\"-Dsystem-freetype=true\" --config-settings=setup-args=\"-Dsystem-qhull=true\""
```
indentation is broken here

bbhtt commented:
You can pass the manifest through a JSON indentor to fix all the indentation or `json.dump` with `indent=4`

itprojects commented:
Pushed new code, with indentation fix, updated deps, compiling in specific sequence (and setting), with the latest tested libraries. Squashed updates.

bbhtt commented:
bot, build io.github.itprojects.MasVisGtk

bbhtt commented:
It built fine but you have lots of appstream related errors in metainfo:

```json
{
    "errors": [
        "appstream-missing-screenshots",
        "appstream-id-mismatch-flatpak-id",
        "appstream-failed-validation"
    ],
    "appstream": [
        "E: io.github.itprojects.MasVisGtk.desktop:36: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:69: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:43: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:71: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:23: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:60: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:33: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:34: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:66: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:53: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:56: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:45: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:~: content-rating-missing",
        "E: io.github.itprojects.MasVisGtk.desktop:25: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:30: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:40: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:49: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:31: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:24: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:50: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:44: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:63: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:26: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:38: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:35: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:65: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:59: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:51: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:48: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:64: description-enum-group-translated description/ul",
        "W: io.github.itprojects.MasVisGtk.desktop:73: screenshot-default-missing",
        "E: io.github.itprojects.MasVisGtk.desktop:54: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:58: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:39: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:55: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:68: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:61: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:28: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:46: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:70: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:29: description-enum-group-translated description/ul",
        "E: io.github.itprojects.MasVisGtk.desktop:41: description-enum-group-translated description/ul"
    ],
    "info": [
        "appstream-missing-screenshots: Catalogue file has no screenshots. Please check if screenshot URLs are reachable",
        "appstream-failed-validation: Metainfo file /tmp/tmpdd_0u7f9/metainfo/io.github.itprojects.MasVisGtk.metainfo.xml has failed validation. Please see the errors in appstream block",
        "appstream-id-mismatch-flatpak-id: The value of ID tag: io.github.itprojects.MasVisGtk.desktop in Metainfo does not match the FLATPAK_ID: io.github.itprojects.MasVisGtk. Please see the docs for more details"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}
```

bbhtt commented:
1. Drop `.desktop` from here https://github.com/itprojects/MasVisGtk/blob/0b1572a79f9135f543ad72aa46810ce2b5121704/data/io.github.itprojects.MasVisGtk.metainfo.xml.in#L3 id tag == Flatapk ID.
2. Screenshots needs to be direct links to PNGs. Tie them to a commit not the main branch.

https://raw.githubusercontent.com/itprojects/MasVisGtk/0b1572a79f9135f543ad72aa46810ce2b5121704/img/1.png

4. The first screenshot needs a `type="default"`, see https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#screenshots
5. Add OARS https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#oars-information

6. Your translations are translating `ul` tags in description. It should be by each `li` and `p` tag instead See https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#metainfo-translations

itprojects commented:
Fixing.

itprojects commented:
Can you point to metadata file for example, to see it in action?

bbhtt commented:
Nope, that looks like a bug in the docs. I'll fix that.

The issue with your description is `ul` items should be under li tags, see https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#description

```
      <ul>
        <li>Waveform plotting</li>
      </ul>
```

whereas you have them as:

```
      <ul>Overview of files (by-folder support)</ul>
      <ul>Frequency plot</ul>
      <ul>Dynamic Range estimation</ul>
      <ul>Peak estimation</ul>
      <ul>Crest Factor estimation</ul>
      <ul>Histogram plot</ul>
      <ul>Peak vs. RMS plot</ul>
      <ul>Loudness analysis, EBU R128 plot</ul>
      <ul>Multi-channel audio support</ul>
```

itprojects commented:
Also, if we're fixing the docs, please consider a non-white and non-black background (like Sepia). it would help greatly!!

bbhtt commented:
You can open an issue on https://github.com/flathub-infra/documentation for design/CSS changes, we use the default docusaurus theme afaik.

itprojects commented:
New push ready for testing.

itprojects commented:
Can the build bot try to build the new (rebased and updated) code, @bbhtt , @hfiguiere ?

bbhtt commented:
You haven't pushed anything here?

bbhtt commented:
bot, build io.github.itprojects.MasVisGtk

itprojects commented:
> You haven't pushed anything here?

Yes, it was just code changes that failed the earlier build.

bbhtt commented:
You should create a new tag, reusing the old tag is a bad practice.

bbhtt commented:
No icon in about:

![image](https://github.com/user-attachments/assets/bf9fbcd5-2b04-4296-a3db-8133052e5a30)

hfiguiere commented on io.github.itprojects.MasVisGtk.json:
```diff
@@ -0,0 +1,102 @@
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/itprojects/MasVisGtk.git",
+                    "tag": "v3.0.0"
```
add the commit that matches the tag here. this protects against retagging.

itprojects replied:
What would happen on the server if the tag is re-tagged?

hfiguiere replied:
you shouldn't be retagging - it's a bad practice (and nothing to do with flatpak), so yeah the build will fail and that's the intent.

itprojects replied:
Yet, if there is no first/initial flathub repository for this app, then that would be acceptable for testing purposes?

hfiguiere replied:
then use a commit and tag when it's ready.

hfiguiere replied:
again this has nothing to do with flatpak. It's a bad practice overall. A published tag SHOULD be IMMUTABLE.

itprojects replied:
Added a commit.

itprojects commented:
@bbhtt , Yes, that is a bug. It wasn't in the first code submission. Changing the icon file does not seem to fix it. Could this be related to the namespace?

bbhtt commented:
> @bbhtt , Yes, that is a bug. It wasn't in the first code submission. Changing the icon file does not seem to fix it. Could this be related to the namespace?

You have a typo here https://github.com/itprojects/MasVisGtk/blob/ca8cb2dcea60899f213db918593342474fa5a59b/src/main_gtk_window.py#L454

itprojects commented:
Oh, YES! That was it.

itprojects commented:
Ready for a new build, @bbhtt ?

itprojects commented:
Compiling `numpy>=2.0`, `scipy`, and `matplotlib` is difficult. It may be helpful, for other projects, to know this line makes all the difference:

```
"pip3 install --no-index --ignore-installed --find-links=file://${PWD} --prefix=${FLATPAK_DEST} --no-build-isolation ."
```

itprojects commented:
@hfiguiere  Is the changed `json` file correct with the commit-only?

bbhtt commented:
Yes

bot, build io.github.itprojects.MasVisGtk

itprojects commented:
@bbhtt  The x86 and aarch64 builds have completed, are there any plans to support RISCV64?

bbhtt commented:
No
